pgo-scheduler/pgo-scheduler.go

```diff
@@ -138,11 +136,21 @@ func main() {
 
 	log.WithFields(log.Fields{}).Infof("Watching namespaces: %s", namespaceList)
 
-	for _, namespace := range namespaceList {
-		SetupWatch(namespace, scheduler, stop)
+	controllerManager, err := sched.NewControllerManager(namespaceList, scheduler)
+	if err != nil {
+		log.WithFields(log.Fields{}).Fatalf("Failed to create controller manager: %s", err)
+		os.Exit(2)
 	}
+	controllerManager.RunAll()
 
-	SetupNamespaceWatch(installationName, scheduler, stop)
+	nsKubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, 0)
+	if err != nil {
```

Comment:
This `err` was already checked on line L140. What should be checked after calling `kubeinformers.NewSharedInformerFactory` ?